### PR TITLE
Storage Drives code is committed in Development branch.

### PR DESCRIPTION
**Pull request by Aliza05**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+
+CLIENT_SECRET_FILE = 'credentials.json'
+API_NAME = 'drive'
+API_VERSION = 'v3'
+SCOPES = ['https://www.googleapis.com/auth/drive']
+
+service = Create_Service(CLIENT_SECRET_FILE, API_NAME, API_VERSION, SCOPES)
+print(dir(service))
+'''
```
Comment (hatim523) [inline main.py]: Missing eof.

File: quickstart.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        print('Files:')
+        for item in items:
+            print(u'{0} ({1})'.format(item['name'], item['id']))
+    except HttpError as error:
+        # TODO(developer) - Handle errors from drive API.
+        print(f'An error occurred: {error}')
+
+
+if __name__ == '__main__':
+    main()
```
Comment (hatim523) [inline quickstart.py]: Why do we have two `main` entry points in a single program?

File: googledrive.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        # Location of the client secrets.
+        CLIENT_SECRETS = 'credentials.json'
+        # Path to the file to upload.
+        FILENAME = file_name
+        # Metadata about the file.
+        MIMETYPE = 'text/plain'
+        TITLE = os.path.basename(file_name)
+        DESCRIPTION = f'A text document about {Path(file_name).stem}.'
+        # Perform OAuth2.0 authorization flow.
+        flow = oauth2client.client.flow_from_clientsecrets(
```
Comment (hatim523) [inline googledrive.py]: Shouldn't the variables needed for initializing `client` reside in the constructor of the class.

File: googledrive.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        flow.redirect_uri = oauth2client.client.OOB_CALLBACK_URN
+        authorize_url = flow.step1_get_authorize_url()
+        print('Go to the following link in your browser: ' + authorize_url)
+        # `six` library supports Python2 and Python3 without redefining builtin input()
+        code = six.moves.input('Enter verification code: ').strip()
+        credentials = flow.step2_exchange(code)
+        # Create an authorized Drive API client.
+        http = httplib2.Http()
+        credentials.authorize(http)
+        drive_service = build('drive', 'v2', http=http)
```
Comment (hatim523) [inline googledrive.py]: Service should only be built once and used as an attribute in different class methods.

File: googledrive.py [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+    def search(self):
+        """Search file in drive location
+
+        Load pre-authorized user credentials from the environment.
+        TODO(developer) - See https://developers.google.com/identity
+        for guides on implementing OAuth2 for the application.
+        """
+        if os.path.exists('token.json'):
+            creds = Credentials.from_authorized_user_file('token.json', SCOPES)
```
Comment (hatim523) [inline googledrive.py]: Redundant code to build the same service every time.

File: googledrive.py [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+                if page_token is None:
+                    break
+
+        except HttpError as error:
+            print(F'An error occurred: {error}')
+            files = None
+
+        return files
+
+    if __name__ == '__main__':
```
Comment (hatim523) [inline googledrive.py]: `main` should not be here since this file should only contain the class code.

File: googledrive.py [hunk 7]
```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/python
```
Comment (hatim523) [inline googledrive.py]: Remove these comments if they are not adding any value to the code below.

File: googledrive.py [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+from google.auth.transport.requests import Request
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+
+# If modifying these scopes, delete the file token.json.
+SCOPES = ['https://www.googleapis.com/auth/drive.metadata.readonly']
+# If modifying these scopes, delete the file token.json.
+SCOPES1 = ['https://www.googleapis.com/auth/drive.readonly']
```
Comment (hatim523) [inline googledrive.py]: How is `SCOPES1` different from `SCOPES`?
What is the difference between these two scopes? Variable names should be self-sufficient in defining their usage.

File: main.py [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+#driver.search()
+file = input("Enter file id: ")
+driver.download(file)
+
+
+
+
+
+'''
+from __future__ import print_function
```
Comment (hatim523) [inline main.py]: Remove this if not needed.

File: storagedrive.py [hunk 10]
```diff
@@ -0,0 +1,17 @@
+from abc import ABC, abstractmethod
+
+class StorageDrive(ABC):
+
+    @abstractmethod
+    def upload(self):
```
Comment (hatim523) [inline storagedrive.py]: `upload` function should define an argument such as filename etc, which should be the basis for input for it's derived classes.

Comment (hatim523) [reply storagedrive.py]: Similarly for `download` and `search` function.

**Review by hatim523 — CHANGES_REQUESTED** [review]
Aliza, have requested some changes in the code, please let me know if you need any help with it.
Thanks :)

**Comment by Aliza05** [comment]
Creating New pull request.

**Pull request closed without merging**